src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first = null;
+    private Node<T> last = null;
+
+    public MyLinkedList() {
+    }
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+            last = newNode;
+        }
+        if (size > 0) {
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            indexCheck(index);
+            Node<T> nodeByIndex = getNode(index);
+            Node<T> newNode = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+            if (index > 0) {
+                nodeByIndex.prev.next = newNode;
+            } else {
+                first = newNode;
+            }
+            nodeByIndex.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        T oldValue = getNode(index).value;
+        getNode(index).value = value;
```

Comment:
```suggestion
        Node<T> nodeByIndex =  getNode(index);
        T oldValue = nodeByIndex.value;
        nodeByIndex.value = value;
```
the method getNode is invoked twice, instead just create a variable for it